### ISSUE TITLE
F1120 22:05:28.240000 34447 net.cpp:717] Check failed: target_blobs[j]->channels() == source_layer.blobs(j).channels() (1 vs. 3) Incompatible parameter size for layer conv1_1

### ISSUE THREAD

**Issue by makaspacex** [TheLegendAli/DeepLab-Context#11]
Training1 net voc12/vgg128_noup
Running ./build/tools/caffe train --solver=voc12/config/vgg128_noup/solver_train_aug.prototxt --weights=voc12/model/vgg128_noup/init.caffemodel --gpu=0
I1120 22:05:22.482954 34447 caffe.cpp:102] Use GPU with device ID 0
I1120 22:05:26.820761 34447 caffe.cpp:110] Starting Optimization
I1120 22:05:26.820894 34447 solver.cpp:32] Initializing solver from parameters: 
train_net: "voc12/config/vgg128_noup/train_train_aug.prototxt"
....
....
....
....
....
I1120 22:05:27.910971 34447 solver.cpp:41] Solver scaffolding done.
I1120 22:05:27.910979 34447 caffe.cpp:118] Finetuning from voc12/model/vgg128_noup/init.caffemodel
I1120 22:05:28.239959 34447 net.cpp:707] Copying source layer conv1_1
F1120 22:05:28.240000 34447 net.cpp:717] Check failed: target_blobs[j]->channels() == source_layer.blobs(j).channels() (1 vs. 3) Incompatible parameter size for layer conv1_1
*** Check failure stack trace: ***
    @     0x7fe28b87aa5d  google::LogMessage::Fail()
    @     0x7fe28b87eef7  google::LogMessage::SendToLog()
    @     0x7fe28b87cd59  google::LogMessage::Flush()
    @     0x7fe28b87d05d  google::LogMessageFatal::~LogMessageFatal()
    @           0x5946d0  caffe::Net<>::CopyTrainedLayersFrom()
    @           0x594df3  caffe::Net<>::CopyTrainedLayersFrom()
    @           0x56c301  train()
    @           0x56e88f  main
    @       0x30f8a1ed1d  (unknown)
    @           0x56b8a9  (unknown)

Process finished with exit code 0

please help me ,thanks

**Comment by TheLegendAli** [comment]
This means Deeplab expected an RGB input, but it looks like your source is grayscale

**Comment by makaspacex** [comment]
@TheLegendAli Thanks, the problem may be happened at data source

**Comment by wolf943134497** [comment]
@zhuihunmiling @TheLegendAli  I have similar error.Did you solve it?My data source download from VOC 2012 official website
I0104 17:21:34.282826 31297 net.cpp:208] This network produces output accuracy
I0104 17:21:34.282845 31297 net.cpp:467] Collecting Learning Rate and Weight Decay.
I0104 17:21:34.282855 31297 net.cpp:219] Network initialization done.
I0104 17:21:34.282858 31297 net.cpp:220] Memory required for data: 9170170576
I0104 17:21:34.282920 31297 solver.cpp:41] Solver scaffolding done.
I0104 17:21:34.282927 31297 caffe.cpp:118] Finetuning from voc12/model/vgg128_noup/init.caffemodel
I0104 17:21:34.386422 31297 net.cpp:740] Target layer fc8_voc12 not initialized.
I0104 17:21:34.387146 31297 solver.cpp:160] Solving vgg128_noup
I0104 17:21:34.387156 31297 solver.cpp:161] Learning Rate Policy: step
E0104 17:21:34.854878 31306 io.cpp:76] Could not open or find file /home/gwm/VOCdevkit/VOC2012/SegmentationClassAug/2011_000541.png
F0104 17:21:34.855020 31306 data_transformer.cpp:332] Check failed: img_height == seg_height (326 vs. 0) 
*** Check failure stack trace: ***
    @     0x7f41d54d8daa  (unknown)
    @     0x7f41d54d8ce4  (unknown)
    @     0x7f41d54d86e6  (unknown)
    @     0x7f41d54db687  (unknown)
    @           0x4dcc6e  caffe::DataTransformer<>::TransformImgAndSeg()
    @           0x59b88d  caffe::ImageSegDataLayer<>::InternalThreadEntry()
    @     0x7f41d3160829  thread_proxy
    @     0x7f41d24f8184  start_thread
    @     0x7f41d222537d  (unknown)
    @              (nil)  (unknown)
Aborted (core dumped)

**Comment by yghlc** [comment]
@zhuihunmiling  I have the same problem as yours, how did you solve the problem?   I didn't change anything in the codes and data.
error information: 
I0409 20:09:39.353346 10655 caffe.cpp:118] Finetuning from voc12/model/vgg128_noup/init.caffemodel
F0409 20:09:39.462594 10655 net.cpp:717] Check failed: target_blobs[j]->channels() == source_layer.blobs(j).channels() (1 vs. 3) Incompatible parameter size for layer conv1_1

More interesting thing is that I have two GPUs with 8GB memory on my computer, but when I used another GPU, the error message was different. like below:

I0409 20:32:57.150202 16472 layer_factory.hpp:78] Creating layer conv4_2
I0409 20:32:57.150223 16472 net.cpp:67] Creating Layer conv4_2
I0409 20:32:57.150239 16472 net.cpp:394] conv4_2 <- conv4_1
I0409 20:32:57.150261 16472 net.cpp:356] conv4_2 -> conv4_2
I0409 20:32:57.150282 16472 net.cpp:96] Setting up conv4_2
    @     0x7f6c1aa8a82d  clone
    @              (nil)  (unknown)
Aborted (core dumped)
